tests/conftest.py

```diff
@@ -58,6 +61,26 @@ def reset_environment():
             os.environ[key] = init_env[key]
 
 
+def remove_non_pytest_log_handlers(logger, *classes):
+    for handler in non_pytest_handlers(logger.handlers):
+        logger.removeHandler(handler)
+
+
+def remove_all_non_pytest_log_handlers():
+    # Remove handlers from the root logger
+    remove_non_pytest_log_handlers(logging.getLogger())
+    # Remove handlers from all other loggers
+    for logger_name in logging.root.manager.loggerDict:
+        remove_non_pytest_log_handlers(logging.getLogger(logger_name))
+
+
+@pytest.fixture
+def clear_all_logger_handlers():
```

Comment:
Should not we set `autouse=True` so that this fixture is automatically applied to all tests? These failures are quite intermittent so I think it can be easy to miss some of them (and also hard to anticipate, in my opinion)

Reply:
I thought about it - but there are just a few tests that are modiyfing handlers, and adding autouse will slow down the whole test suite.  So I am not sure if this is a good idea. But maybe it's worth to do it indeed ? @dstandish - WDYT?

Reply:
But other than that, LGTM :) And good job finding this annoying weird error!

Reply:
yeah. I will merge it now - and we can always change it later.